backend/src/friend/friend.service.ts

```diff
@@ -1,4 +1,452 @@
-import { Injectable } from '@nestjs/common';
+import {
+  Injectable,
+  Res,
+  BadRequestException,
+  ServiceUnavailableException,
+} from '@nestjs/common';
+import { User } from '@prisma/client';
+import { PrismaService } from 'nestjs-prisma';
+import { Response } from 'express';
 
 @Injectable()
-export class FriendService {}
+export class FriendService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  /*****************************************************************************/
+  /* send friend Request */
+  /*****************************************************************************/
+  async sendFriendRequest(
+    user: User,
+    payload: { userEmail: string },
+    res: Response,
+  ) {
+    // check user/payload data
+    const checkData = await this.checkSenderAndReceiver(user, payload);
+    if (checkData !== '') {
+      throw new BadRequestException(checkData);
+    }
+
+    const senderEmail: string = user.email;
+    const receiverEmail: string = payload.userEmail;
+
+    // find the receiver in database
+    const sender: User = user;
+    const receiver: User = await this.getUserByEmail(receiverEmail);
+    if (!receiver) {
+      throw new BadRequestException('Receiver not found.');
+    }
+
+    // check if receiver's receivedRequest array contains the sender's email
+    if (receiver.friendRequestsReceived.includes(senderEmail)) {
+      throw new BadRequestException(
+        "Friend request already in receiver's list",
+      );
+    }
+
+    // check if sender's sentRequest array contains the receiver's email
+    if (sender.friendRequestsSent.includes(receiverEmail)) {
+      throw new BadRequestException("Friend request already in sender's list.");
+    }
+
+    // check if receiver was the sender
+    if (
+      receiver.friendRequestsSent.includes(senderEmail) ||
+      sender.friendRequestsReceived.includes(receiverEmail)
+    ) {
+      throw new BadRequestException('The receiver sent you an request.');
+    }
+
```

Comment:
I don't understand what you are checking here. Because you already verified the sender is not inviting themselves in the checkSenderAndReceiver function.